src/codeu/chat/server/Server.java

```diff
@@ -201,6 +207,76 @@ public void run() {
     });
   }
 
+  private void reloadOldData() throws IOException {
+    // Open the transaction log file for reading
+    FileReader fileReader = new FileReader("data/transaction_log.txt");
+    BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+    // Read the header lines of each transaction log
+    String line = bufferedReader.readLine();
+
+    while(line != null) {
+      String[] logInfo = line.split(" ");
```

Comment:
You can probably use @JiahuiKChen's `Tokenizer` class here which would simplify some of the parsing down below. You might have to put the timestamp earlier in the data sequence though.